src/system/Nav/NavItem.tsx

```diff
@@ -38,52 +41,91 @@ const NavItemBase = forwardRef< HTMLLIElement, NavItemBaseProps >(
 	)
 );
 
-const NavLink = forwardRef< HTMLAnchorElement, NavItemProps >(
-	(
-		{ children, href, active, disabled, variant = 'primary', ...props }: NavItemProps,
-		ref: Ref< HTMLAnchorElement >
-	) => (
-		<NavigationMenu.Link
-			className={ classNames( `${ VIP_NAV }-item-link` ) }
-			href={ href }
-			ref={ ref }
-			sx={ navItemStyles( variant ) }
-			active={ active }
-			aria-current={ active ? 'page' : undefined }
-			aria-disabled={ disabled }
-			{ ...props }
-		>
-			{ children }
-		</NavigationMenu.Link>
-	)
-);
-
 export interface NavItemProps extends NavigationMenu.NavigationMenuLinkProps {
 	className?: string;
 	disabled?: boolean;
 	variant?: NavVariant;
+	icon?: JSX.Element;
+	render?: React.FC< { href?: string; children?: React.ReactNode } >;
```

Comment:
This is conflating render props and component props in a confusing way that will lead to issues. It's named like a render prop (a JavaScript function that returns JSX—not actually a component!), but it's implemented (sort of) like a component prop (a component definition that will be invoked in a JSX expression).

They are very similar in practice, and which to use probably comes down to how often the design system consumer will need to customize the props passed to the foreign link component. For example, do you expect the consumer to simply pass the Next.js `Link` component without need for further customization? If so, use a component prop, but use PascalCase to indicate that the prop expects a component definition. Also type the prop using `React.ComponentType` (don't artificially limit it to `React.FC`):

```ts
import Link from 'next/link';

<NavItem.Menu LinkComponent={ Link }>Settings</NavItem.Menu>
```

On the other hand, maybe you expect the consumer to need customization frequently (e.g., additional props passed to `Link`). If so, use a render prop, but name it more descriptively and type it as a plain function (not as a component). This will avoid the need for the consumer to create one-off components.

```ts
import Link from 'next/link';

<NavItem.Menu
   renderLink={
      ( href: string, children?: ReactNode ) => <Link children={ children } href={ href } scroll={ false } />
   }
>Settings</NavItem.Menu>
```

I will say that, in the render prop example, the "teleporting" of the `children` prop feels a little strange and unnecessary. It can be worked around, though.

The component prop pattern could be a much more straightforward implementation if you give it a default value (i.e., a generic HTML link component implementation).